src/map/storage.c

```diff
@@ -95,37 +97,104 @@ void do_reconnect_storage(void)
 	gstorage->db->foreach(gstorage->db, storage->reconnect_sub);
 }
 
+/**
+ * Get a storage id by its name (through @commands etc...)
+ * @param[in] storage_name     pointer to the storage name char array.
+ * @return id of the storage or -1 if not found.
+ */
+int storage_get_id_by_name(const char *storage_name)
+{
+	int i = 0;
+
+	ARR_FIND(0, VECTOR_LENGTH(storage->configuration), i, strcmp(VECTOR_INDEX(storage->configuration, i).name, storage_name) == 0);
+
+	if (i == VECTOR_LENGTH(storage->configuration))
+		return -1;
+
+	return VECTOR_INDEX(storage->configuration, i).uid;
+}
+
+/**
+ * Get storage with a particular ID from a player.
+ * @param[in] sd         pointer to map session data.
+ * @param[in] storage_id ID of the storage to receive.
+ * @return pointer to player's storage data structure or null if not found.
+ */
+struct storage_data *storage_ensure(struct map_session_data *sd, int storage_id)
+{
+	int i = 0;
+	struct storage_data *stor = NULL;
+
+	nullpo_retr(NULL, sd);
```

Comment:
Here need return not NULL but some kind of default storage.
Because in other code no checks for return from storage->ensure